lib/utils.pm

```diff
@@ -533,7 +533,13 @@ sub zypper_call {
 
     unless (grep { $_ == $ret } @$allow_exit_codes) {
         upload_logs('/var/log/zypper.log');
-        die "'zypper -n $command' failed with code $ret";
+        my $msg = "'zypper -n $command' failed with code $ret";
+        if ($ret == 104) {
+            $msg .= " (ZYPPER_EXIT_INF_CAP_NOT_FOUND)\n\nRelated zypper logs:\n";
+            script_run('tac /var/log/zypper.log | grep -F -m1 -B10000 "Hi, me zypper" | tac | grep \'\(SolverRequester.cc\|THROW\|CAUGHT\)\' > /tmp/z104.txt');
```

Comment:
One thing I missed, is `tac` available in older systems? as this might affect maintenance

Reply:
I would guess it is. It is an old part (copyright in sourcecode sais 1988) of coreutils.

Also I found it in this PTF: https://build.suse.de/package/binaries/PTF:9981/coreutils.SUSE_SLE-11_GA/SUSE_SLE-11_GA - that should be old enough I guess.